app/src/main/java/org/covidwatch/android/data/contactevent/ContactEventsDownloader.kt

```diff
@@ -34,17 +34,19 @@ class ContactEventsDownloader(private val workManager: WorkManager) {
             downloadRequest
         )
 
-        workManager.getWorkInfosForUniqueWorkLiveData(ONE_TIME_REFRESH)
-            .observe(lifecycle, Observer { workInfos ->
-                val workInfo = workInfos.firstOrNull()
-                if (workInfo != null) {
-                    when {
-                        workInfo.state == WorkInfo.State.SUCCEEDED -> callback(true)
-                        workInfo.state == WorkInfo.State.FAILED -> callback(false)
-                        workInfo.state == WorkInfo.State.CANCELLED -> callback(false)
-                    }
+        val stateLiveData = MediatorLiveData<Boolean>()
+        stateLiveData.addSource(workManager.getWorkInfosForUniqueWorkLiveData(ONE_TIME_REFRESH)) {
```

Comment:
Thanks for catching that returned LiveData from `enqueueUniqueWork` doesn't work as I expected.

But I still have concerns about this implementation:
Why not just to return `getWorkInfosForUniqueWorkLiveData(ONE_TIME_REFRESH)` livedata? 
Mapping all values to `true` leaves us without the possibility to handle different cases in the clients then. Taking into account that we don't care or handle this case in the UI we could just listen to any value and react the same way to all of them.

Also, with this logic we could use just `map` instead of `MediatorLiveData`. Or there were some problems with `map`? I know that sometimes it tricky with start of observation and when the real value is ready so it doesn't trigger in the end.

Reply:
I'm currently not mapping all values but just the ones that correspond to "work finished"
Also, not returning the original LiveData as it is a list of all WorkInfos with that unique name. As a whole I think this is exposing implementation details instead of just returning a LiveData that lets you know when your work is finished. You don't need to know about WorkManager and won't care if it changes in the future.

Reply:
My bad, the values correspond to finished work, yes. But I still don't understand why we need `MediatorLiveData` there instead of just map transformation?

I double checked what API we have to get work info: Why not to use `getWorkInfoByIdLiveData` that gives you only the state of particular work request?

```
return workManager.getWorkInfoByIdLiveData(downloadRequest.id)
```

I don't think returning WorkInfo is exposing any implementation. This gives the flexibility to handle it by clients how it's convenient for their use cases. For instance, now, there is no case for `BLOCKED` state which happens when there is no internet. The pull refresh won't stop until a user find a connection. 

With above return statement from `executePublicContactEventsRefresh`
we get this code in `HomeViewModel`
```
_isRefreshing.addSource(state) {
    when (it.state) {
        WorkInfo.State.SUCCEEDED,
        WorkInfo.State.FAILED,
        WorkInfo.State.CANCELLED -> {
            _isRefreshing.value = false
            _isRefreshing.removeSource(state)
        }
        WorkInfo.State.BLOCKED -> {
            // Hide refresh notify user about no connection
        }
        else -> Unit
    }
}
```

Reply:
Actually BLOCKED state refers to the situation where there is preceding enqueued work that is not finished, not when the conditions are not (yet) satisfied. Unfortunately, in this situation the work is considered enqueued which doesn't tell us much. Which is a good argument to not use WorkManager for user-requested operations. Which is why I don't like to expose the underlying implementation through WorkManager as we might decide to change it later. However, for such a minor feature that will be used mainly by our demonstrators I don't think it's worth the effort. At least not in this stage.
About the other note - yeah, getting the work info by id is the logical way to do it. I should've checked that. Added a commit.

Reply:
From the docs about the `BLOCKED`:
>Used to indicate that the WorkRequest is currently blocked because its prerequisites haven't finished successfully. 

That `prerequisites` part confused me but I tried again to see if it returns BLOCKED if there is no internet and it doesn't work as I expected.

Anyways, I finally understood what you meant by exposing `WorkInfo`. Our own abstraction on this API is needed and a simple `true\false` is more than enough for this demo task.

Good job :+1: